vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java

```diff
@@ -0,0 +1,64 @@
+package com.vaadin.flow.spring.security;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+
+import com.vaadin.flow.server.HandlerHelper;
+import com.vaadin.flow.server.connect.EndpointUtil;
+import com.vaadin.flow.spring.VaadinConfigurationProperties;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+/**
+ * Contains utility methods related to request handling.
+ */
+@Component
+public class RequestUtil {
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    private VaadinConfigurationProperties configurationProperties;
+
+    private Object endpointUtil;
+
+    @PostConstruct
+    public void init() {
+        try {
+            endpointUtil = applicationContext.getBean(EndpointUtil.class);
+        } catch (NoClassDefFoundError e) {
+            // Presumable Fusion is not on the classpath
+        } catch (Exception e) {
+            // Presumable Fusion is not on the classpath
+        }
+
+    }
```

Comment:
Would a cleaner solution be injecting 
```
@Autowired(required=false)
private EndpointUtil endpointUtil;
```

Reply:
That was the initial solution. Then it was changed to this to support running without Fusion on the classpath

Reply:
`required=false` still requires the class to be present

Reply:
Copy paste a previous reply from Artur 

> It results in Caused by: java.lang.IllegalStateException: Failed to introspect Class [com.vaadin.flow.spring.security.VaadinDefaultRequestCache] from ClassLoader [sun.misc.Launcher$AppClassLoader@18b4aac2]

Reply:
Right. My concern was if there is risk that exceptions in bean creation not being distinguishable from bean missing. But I guess the exception chain is printed anyway in either case.